### PR TITLE
[Dev Portal] Authenticator listing component for IDP and Application views.

### PR DESCRIPTION
**Pull request by brionmario**
## Purpose
Authenticators could be easily managed through the UI and consistent experience should be there for both IDP and Application views. 

## Goals
This PR introduces the following reusable components.

1. `SegmentedAccordion` - `@wso2is/react-components`
2. `AuthenticatorAccordion` - `components/shared`

## Approach

**Usage**

```tsx
<AuthenticatorAccordion
  globalActions={[
    {
      icon: 'trash alternate',
      onClick: (e: MouseEvent<HTMLDivElement>, id: string): void => {
        setShowDeleteConfirmationModal(true);
        setDeletingAuthenticator(
          federatedAuthenticators.authenticators.find(
            (authenticator) => authenticator.authenticatorId === id
          )
        );
      },
      type: 'icon',
    },
  ]}
  authenticators={[
    {
      actions: [
        {
          defaultChecked: true,
          label: 'Make default',
          onChange: handleDefaultAuthenticatorChange,
          type: 'checkbox',
        },
        {
          defaultChecked: true,
          label: 'Enabled',
          onChange: handleAuthenticatorEnableToggle,
          type: 'toggle',
        },
      ],
      content: federatedAuthenticators.defaultAuthenticatorId && (
        <AuthenticatorFormFactory
          metadata={authenticatorMeta}
          initialValues={authenticatorDetails}
          onSubmit={handleInboundConfigFormSubmit}
          type={authenticatorMeta?.name}
        />
      ),
      id: authenticatorDetails?.authenticatorId,
      title: authenticatorDetails?.name,
    },
  ]}
/>;
```

**Screenshots**

Screen Shot 2020-04-17 at 6 03 30 AM

Screen Shot 2020-04-17 at 10 28 30 AM

### CODE REVIEW

File: apps/admin-portal/src/components/identity-providers/settings/authenticator-settings.tsx [hunk 1]
```diff
@@ -175,27 +180,130 @@ export const AuthenticatorSettings: FunctionComponent<IdentityProviderSettingsPr
...
+     * @param {React.FormEvent<HTMLInputElement>} e - Event.
+     * @param {CheckboxProps} data - Checkbox data.
+     * @param {string} id - Id of the authenticator.
+     */
+    const handleDefaultAuthenticatorChange = (
+        e: FormEvent<HTMLInputElement>,
+        data: CheckboxProps,
+        id: string): void => {
+
+        // Implement necessary logic here.
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/identity-providers/settings/authenticator-settings.tsx:195]:
```suggestion
        // TODO: Implement necessary logic here.
```

File: apps/admin-portal/src/components/identity-providers/settings/authenticator-settings.tsx [hunk 2]
```diff
@@ -175,27 +180,130 @@ export const AuthenticatorSettings: FunctionComponent<IdentityProviderSettingsPr
...
+
+    /**
+     * Handles authenticator enable toggle.
+     *
+     * @param {React.FormEvent<HTMLInputElement>} e - Event.
+     * @param {CheckboxProps} data - Checkbox data.
+     * @param {string} id - Id of the authenticator.
+     */
+    const handleAuthenticatorEnableToggle = (e: FormEvent<HTMLInputElement>, data: CheckboxProps, id: string): void => {
+        // Implement necessary logic here.
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/identity-providers/settings/authenticator-settings.tsx:206]:
```suggestion
        // TODO: Implement necessary logic here.
```

Comment (brionmario) [reply apps/admin-portal/src/components/identity-providers/settings/authenticator-settings.tsx]: Fixed with `2b41044`

File: apps/admin-portal/src/components/identity-providers/settings/authenticator-settings.tsx (hunk 1 above)
Comment (brionmario) [reply apps/admin-portal/src/components/identity-providers/settings/authenticator-settings.tsx]: Fixed with `2b41044`

File: modules/theme/src/theme-core/definitions/globals/product.less [hunk 3]
```diff
@@ -62,6 +62,10 @@ body {
          Global Helper Classes
```
Comment (jeradrutnam) [inline modules/theme/src/theme-core/definitions/globals/product.less:62]: We'll move this section to a new file. may be `helpers` in `globals`. not in this, in a separate PR

File: modules/theme/src/themes/default/globals/site.variables [hunk 4]
```diff
@@ -67,6 +67,7 @@
 @doveGray         : #666666;
 @dullGreen        : #34a850;
 @paleGray         : #D6D6D6;
+@alabaster        : #FCFCFC;
```
Comment (jeradrutnam) [inline modules/theme/src/themes/default/globals/site.variables:70]:
```suggestion
@paleGray         : #d6d6d6;
@alabaster        : #fcfcfc;
```

File: modules/theme/src/theme-core/definitions/globals/product.less (hunk 3 above)
Comment (brionmario) [reply modules/theme/src/theme-core/definitions/globals/product.less:62]: +1. Lets do that.

**Pull request merged**